aiohomekit/controller/ble/pairing.py

```diff
@@ -851,17 +879,17 @@ async def _populate_accessories_and_characteristics(
             if self._shutdown:
                 return
 
-            was_connected = self.client and self.client.is_connected
-            self._restore_pending |= not was_connected
             self._tried_to_connect_once = True
+
+            made_connection = await self._ensure_connected(attempts)
```

Comment:
I made ensure_connected return if it had to make a connection or not since there was a race here since we didn't hold the lock while we checked for `was_connected` above